px4_ros2_cpp/include/px4_ros2/navigation/experimental/local_navigation_interface.hpp

```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <optional>
+#include <rclcpp/rclcpp.hpp>
+#include <Eigen/Eigen>
+
+#include <px4_msgs/msg/vehicle_odometry.hpp>
+#include <px4_ros2/navigation/experimental/navigation_interface_base.hpp>
+
+namespace px4_ros2
+{
+
+using AuxLocalPosition = px4_msgs::msg::VehicleOdometry;
+
+enum class PoseFrame
+{
+  Unknown,
+  NED,
+  FRD
+};
+
+enum class VelocityFrame
+{
+  Unknown,
+  NED,
+  FRD,
+  BodyFRD
+};
+
+constexpr inline uint8_t poseFrameToMessageFrame(px4_ros2::PoseFrame frame) noexcept
+{
+  switch (frame) {
+    case PoseFrame::Unknown: return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+
+    case PoseFrame::NED: return AuxLocalPosition::POSE_FRAME_NED;
+
+    case PoseFrame::FRD: return AuxLocalPosition::POSE_FRAME_FRD;
+  }
+
+  return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+}
+
+constexpr inline uint8_t velocityFrameToMessageFrame(px4_ros2::VelocityFrame frame) noexcept
+{
+  switch (frame) {
+    case VelocityFrame::Unknown: return AuxLocalPosition::VELOCITY_FRAME_UNKNOWN;
+
+    case VelocityFrame::NED: return AuxLocalPosition::VELOCITY_FRAME_NED;
+
+    case VelocityFrame::FRD: return AuxLocalPosition::VELOCITY_FRAME_FRD;
+
+    case VelocityFrame::BodyFRD: return AuxLocalPosition::VELOCITY_FRAME_BODY_FRD;
+  }
+
+  return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+}
+
+/**
+ * @struct LocalPositionEstimate
```

Comment:
```suggestion
 * @struct LocalPositionMeasurement
```

Nitpicking here, but using the name "Estimate" is a bit confusing here as we should be sending "measurements" to the estimator and getting back "estimates". Another possible shorter name is `LocalPositionSample`.